common/network-types/src/frame.rs

```diff
@@ -346,13 +400,22 @@ impl<'a> FrameReassembler<'a> {
             Entry::Vacant(v) => {
                 // Begin building a new frame
                 v.insert(FrameBuilder::new(segment, ts));
-                self.sorted_seq_ids.insert(seq_id);
+                self.sorted_seq_ids.insert(frame_id);
             }
         }
 
         Ok(())
     }
 
+    /// Returns [information](FrameInfo) about the incomplete frames.
+    /// The ordered frame IDs are the keys on the returned map.
+    pub fn incomplete_frames(&self) -> BTreeMap<FrameId, FrameInfo> {
+        self.sequences
+            .iter()
+            .filter_map(|e| (!e.is_complete()).then(|| (e.frame_id, e.info())))
+            .collect()
+    }
```

Comment:
What are these functions used for? Retransmission? That would need a more detailed view into individual frame's segments.

Reply:
Genau. That more detailed view is in FrameInfo.